tools/build_chain.sh

```diff
@@ -1275,34 +1541,31 @@ for line in ${ip_array[*]};do
         nodeid_list=$"${nodeid_list}node.${count}=${nodeid}
     "
         fi
-        
-        ip_list=$"${ip_list}node.${count}="${ip}:$(( $(get_value ${ip//./}_count) + port_start[0] ))"
+        local node_ports=
+        read -r -a node_ports <<< "${port_start[*]}"
+        if [ -n "${ports_array[server_count]}" ];then 
+            read -r -a node_ports <<< "${ports_array[server_count]//,/ }"
+            echo "node_ports ${node_ports[*]}" >>"${logfile}"
+            [ "${i}" == "0" ] && { set_value "${ip//./}_port_offset" 0; }
+        fi
+        ip_list="${ip_list}node.${count}=${ip}:$(( $(get_value ${ip//./}_port_offset) + node_ports[0] ))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5426761)